native broken on OS X

#4058 broke the building on native for OS X:

```console
RIOT/core/sched.c:61:47: error: argument to 'section' attribute is not valid for this target: mach-o section
      specifier requires a segment and section separated by a comma
__attribute__((used)) __attribute__((section (".openocd")))
                                              ^
RIOT/core/sched.c:67:47: error: argument to 'section' attribute is not valid for this target: mach-o section
      specifier requires a segment and section separated by a comma
__attribute__((used)) __attribute__((section (".openocd")))
                                              ^
2 errors generated.
```

@cgundogan is Jenkins now also compiling for OS X?
@kaspar030 What's Murdock state on this?

--- a/core/sched.c
+++ b/core/sched.c
@@ -58,13 +58,19 @@ clist_node_t sched_runqueues[SCHED_PRIO_LEVELS];
 static uint32_t runqueue_bitcache = 0;
 
 /* Needed by OpenOCD to read sched_threads */
-__attribute__((used)) __attribute__((section (".openocd")))
+#if defined(__APPLE__) && defined(__MACH__)
+ #define FORCE_USED_SECTION __attribute__((used)) __attribute__((section ("__OPENOCD,__openocd")))
+#else
+ #define FORCE_USED_SECTION __attribute__((used)) __attribute__((section (".openocd")))
+#endif
+
+FORCE_USED_SECTION
 uint8_t max_threads = sizeof(sched_threads) / sizeof(thread_t*);
 
 #ifdef DEVELHELP
 /* OpenOCD can't determine struct offsets and additionally this member is only
  * available if compiled with DEVELHELP */
-__attribute__((used)) __attribute__((section (".openocd")))
+FORCE_USED_SECTION
 uint8_t _tcb_name_offset = offsetof(thread_t, name);
 #endif
 

